.github/workflows/terraform-observe_scheduler.yaml

```diff
@@ -57,7 +57,32 @@ jobs:
           OBSERVE_DOMAIN: ${{ secrets.TERRAFORM_MODULES_TEST_OBSERVE_DOMAIN }}
           OBSERVE_USER_EMAIL: ${{ secrets.TERRAFORM_MODULES_TEST_OBSERVE_USER_EMAIL }}
           OBSERVE_USER_PASSWORD: ${{ secrets.TERRAFORM_MODULES_TEST_OBSERVE_USER_PASSWORD }}
-          
+
           GITHUB_WORKSPACE: ${{ github.workspace }}
           GITHUB_SHA: ${{ github.sha }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  # Validate App Icon URL is internal
+  validateAppIcon:
+    name: Validate App Icon
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - uses: actions/checkout@v3
+      - name: Check manifest.yaml exists
+        id: check_manifest
+        uses: andstor/file-existence-action@v2
+        with:
+          files: "manifest.yaml"
+      - name: Get App Icon URL
+        id: get_icon_url
+        uses: mikefarah/yq@v4.31.1
+        if: steps.check_manifest.outputs.files_exists == 'true'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          cmd: yq '.iconUrl' manifest.yaml
+      - name: Validate App Icon URL
+        if: ${{ !startsWith(steps.get_icon_url.outputs.result, 'https://assets.observeinc.com/') && steps.check_manifest.outputs.files_exists == 'true' }}
+        run: echo "App icon URLs must live in the 'assets.observeinc.com' domain" && exit 1
```

Comment:
Use command annotations! 

https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-a-warning-message

Ideally you want AST-aware parsing when printing errors like this so you know which line declared the thing you're validating. But at a minimum the error should be reported on the file in question and not buried in Actions logs.